helm/dagger/templates/engine-daemonset.yaml

```diff
@@ -0,0 +1,103 @@
+{{- if .Values.magicache.enabled }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "dagger.fullname" . }}-magicache-token
+  namespace:  {{ .Release.Namespace }}
+data:
+  _EXPERIMENTAL_DAGGER_CACHESERVICE_TOKEN: {{ required "A magicache token is required" .Values.magicache.token | b64enc | quote }}
+{{- end }}
+{{- if .Values.engine.config }}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "dagger.fullname" . }}-engine-config
+  namespace: {{ .Release.Namespace }}
+data:
+  engine.toml: |{{- .Values.engine.config | nindent 4 }}
+{{- end }}
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ include "dagger.fullname" . }}-engine
+  namespace: {{ .Release.Namespace }}
+  labels:
+    k8s-app: {{ include "dagger.fullname" . }}-engine
+spec:
+  selector:
+    matchLabels:
+      name: {{ include "dagger.fullname" . }}-engine
+  template:
+    metadata:
+      labels:
+        name: {{ include "dagger.fullname" . }}-engine
+    spec:
+      {{- with .Values.engine.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.engine.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      securityContext:
+        runAsUser: 0
+        runAsGroup: 1001
+        fsGroup: 1001
+        fsGroupChangePolicy: "OnRootMismatch"
+      containers:
+        - name: dagger-engine
+          image: {{ .Values.engine.image.repository }}:{{ if .Values.engine.image.tag }}{{ .Values.engine.image.tag }}{{ else }}{{ .Chart.AppVersion }}{{ end }}
+          imagePullPolicy: Always
+          args:
+            - "--oci-max-parallelism"
+            - "num-cpu"
+          {{- if .Values.magicache.enabled }}
+          env:
+          - name: _EXPERIMENTAL_DAGGER_CACHESERVICE_URL
+            value: {{ required "A magicache url is required" .Values.magicache.url }}
+          envFrom:
+          - secretRef:
+              name: {{ include "dagger.fullname" . }}-magicache-token
+          {{- end }}
+          securityContext:
+            privileged: true
+            capabilities:
+              add:
+                - ALL
+          resources: {{- toYaml .Values.engine.resources | nindent 12 }}
+          readinessProbe:
+            exec:
+              command: ["buildctl", "debug", "workers"]
```

Comment:
What are your thoughts on following up on this with `grpcurl -plaintext localhost:1234 grpc.health.v1.Health/Check` when we upgrade to a BuildKit that has this feature?

https://github.com/moby/buildkit/pull/4145

Reply:
I'm definitely up for this. Last I had checked... it wasn't available. The only thing that I don't like about it is that we'll need to include `grpcurl` in the image.